src/System.Linq/src/System/Linq/Partition.cs

```diff
@@ -292,39 +267,41 @@ public IPartition<TSource> Take(int count)
                 return new ListPartition<TSource>(_source, _minIndex, (uint)maxIndex >= (uint)_maxIndex ? _maxIndex : maxIndex);
             }
 
-            public TSource ElementAt(int index)
+            public bool TryGetElementAt(int index, out TSource element)
             {
-                if (((uint)index > (uint)_maxIndex - _minIndex) || (index >= _source.Count - _minIndex)) throw Error.ArgumentOutOfRange("index");
-                return _source[_minIndex + index];
-            }
+                if ((uint)index <= (uint)(_maxIndex - _minIndex) && index < _source.Count - _minIndex)
```

Comment:
The mathematical result is the same, but casting just `_maxIndex` is less good as the result of `(uint)_maxIndex - _minIndex` is `long`.

I didn't so much change it on purpose, as fix it by accident by writing what I'd meant to do the first time when I came to rephrase it.